provider/marathon_test.go

```diff
@@ -551,7 +551,7 @@ func TestMarathonTaskFilter(t *testing.T) {
 					},
 				},
 			},
-			expected:         false,
+			expected:         true,
```

Comment:
Instead of flipping the `expected` flag, I think we should remove the entire test case as we have removed the restriction to specify a known port only.